pytorch_pfn_extras/training/extensions/slack.py

```diff
@@ -0,0 +1,205 @@
+import os
+import json
+import warnings
+
+from typing import Any, Callable, List, Optional, Union
+
+from pytorch_pfn_extras.training import extension
+from pytorch_pfn_extras.training import trigger as trigger_module
+from pytorch_pfn_extras.training._manager_protocol import ExtensionsManagerProtocol
+
+try:
+    import requests  # type: ignore[import]
+    _requests_available = True
+except ImportError:
+    _requests_available = False
+
+try:
+    import slack_sdk
+    _slack_sdk_available = True
+except ImportError:
+    _slack_sdk_available = False
+
+
+class Slack(extension.Extension):
+    """An extension to communicate with Slack.
+
+    This extension receives a ``text_template`` argument that contains
+    placeholders that will be populated with ``manager`` or ``manager.observation``
+    fields, and custom objects such as ``context_object``.
+
+    For example. `text_template = "Loss {loss} for iteration {.iteration}"`
+    will be populated as `text_template.format(manager, context_object, **observation)`
+    retrieving the ``loss`` value from the ``observation`` and ``.iteration`` from
+    the manager object. Instead of string, a callable object taking the
+    ``ExtensionsManager``, ``context_object`` and the observation dictionary can
+    be used instead. The same rules apply for the ``filenames_template`` argument.
+
+    Args:
+        channel_id (str): The channel where messages or files will be sent.
+        text_template (str or callable): Template for sending the message.
```

Comment:
Can these field names be simpler, e.g. `msg`, `start_msg`, `end_msg`, `filenames`?

We need to provide a nice default values for start and end message.

```py
_identity = '`{getpass.getuser()}@{socket.gethostname()}` [PID `{os.getpid()}`]'

def _default_start_msg(_, _) -> str:
    return f'''
🏃 **Training started! {_identity}**
Command: `{shlex.quote(sys.argv)}`
'''

_default_end_msg = '✅ **Training finished! {_identity}**'
```

Reply:
Added!